### PR TITLE
8257709: C1: Double assignment in InstructionPrinter::print_stack

### PR DESCRIPTION
**Pull request by sergey-platonov**
There is currently a minor bug (just a typo, actually) in the hotspot, leading
to undefined behavior.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257709](https://bugs.openjdk.java.net/browse/JDK-8257709): C1: Double assignment in InstructionPrinter::print_stack

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to ca66b57a63563666e4b5ebfc216cb6c1a27f51f2
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to ca66b57a63563666e4b5ebfc216cb6c1a27f51f2

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1519/head:pull/1519`
`$ git checkout pull/1519`

### CODE REVIEW

**Comment by shipilev** [comment]
Submitted: https://bugs.openjdk.java.net/browse/JDK-8257709 -- please rename this PR to "8257709: C1: Double assignment in InstructionPrinter::print_stack" to get it hooked properly. Also, can you please go to https://github.com/sergey-platonov/jdk/actions and check that testing workflow is enabled? If not, can you trigger it manually on your branch?

File: src/hotspot/share/c1/c1_InstructionPrinter.cpp [hunk 1]
```diff
@@ -239,7 +239,7 @@ void InstructionPrinter::print_stack(ValueStack* stack) {
     output()->cr();
     fill_to(start_position, ' ');
     output()->print("locks [");
-    for (int i = i = 0; i < stack->locks_size(); i++) {
+    for (int i = 0; i < stack->locks_size(); ++i) {
```
Comment (shipilev) [inline src/hotspot/share/c1/c1_InstructionPrinter.cpp:242]: I would prefer to keep `i++`, as in old code, unless it fixes something I don't quite see.

**Review by shipilev — CHANGES_REQUESTED** [review]

File: src/hotspot/share/c1/c1_InstructionPrinter.cpp (hunk 1 above)
Comment (sergey-platonov) [reply src/hotspot/share/c1/c1_InstructionPrinter.cpp]: Done. Can you, please, create an issue, so I can mention it here?

Comment (shipilev) [reply src/hotspot/share/c1/c1_InstructionPrinter.cpp]: I did already, see https://github.com/openjdk/jdk/pull/1519#issuecomment-738180725.

Comment (sergey-platonov) [reply src/hotspot/share/c1/c1_InstructionPrinter.cpp]: Oh, sorry! I guess, haven't completely got back from the vacation :-)

Comment (sergey-platonov) [reply src/hotspot/share/c1/c1_InstructionPrinter.cpp]: Pipeline is on the way.

**Review by shipilev — APPROVED** [review]
Looks good.

**Comment by shipilev** [comment]
...ah, and maybe merge from `master` to get all the recent fixes. Current tests might fail on known issues. Not that it is necessary for a simple issue like this, but it helps to learn the contribution ropes in this project.

**Review by chhagedorn — APPROVED** [review]
Looks good.

**Comment by sergey-platonov** [comment]
> ...ah, and maybe merge from `master` to get all the recent fixes. Current tests might fail on known issues. Not that it is necessary for a simple issue like this, but it helps to learn the contribution ropes in this project.

Done! Will commits be squashed?

**Comment by shipilev** [comment]
> > ...ah, and maybe merge from `master` to get all the recent fixes. Current tests might fail on known issues. Not that it is necessary for a simple issue like this, but it helps to learn the contribution ropes in this project.
> 
> Done! Will commits be squashed?

Yes, bots would squash the commits during integration. You don't need to rebase or do anything else Git-wise. Once testing is clean, you can say `/integrate`, as bots suggested.

**Comment by sergey-platonov** [comment]
> > > ...ah, and maybe merge from `master` to get all the recent fixes. Current tests might fail on known issues. Not that it is necessary for a simple issue like this, but it helps to learn the contribution ropes in this project.
> > 
> > 
> > Done! Will commits be squashed?
> 
> Yes, bots would squash the commits during integration. You don't need to rebase or do anything else Git-wise. Once testing is clean, you can say `/integrate`, as bots suggested.

Great! Thank you for your help, Aleksey!

**Comment by shipilev** [comment]
If you go an look at "Checks" tab, you will see it says that everything is green. You can say `/integrate` now.

**Pull request closed without merging**